tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/lock/LocksInMainTest.java

```diff
@@ -264,25 +134,29 @@ public void testLockNegativeCases() {
                                   18, 9);
     }
 
-    // https://github.com/ballerina-platform/ballerina-lang/issues/22987
-    @Test(description = "Test for parallel run using locks", enabled = false)
+    @Test()
+    public void testPanicIfInLockConcurrently() {
+        BRunUtil.invoke(parallelCompileResult, "testPanicIfInLockConcurrently");
+    }
+
+    @Test(description = "Test for parallel run using locks")
     public void testParallelRunUsingLocks() {
-        Object returns = BRunUtil.invoke(parallelCompileResult, "runParallelUsingLocks");
+        BRunUtil.invoke(parallelCompileResult, "runParallelUsingLocks");
     }
 
     @Test(description = "Test for parallel run when invocations have global variable dependencies")
     public void testParallelRunWithInvocationDependencies() {
-        Object returns = BRunUtil.invoke(parallelCompileResult, "testLockWithInvokableAccessingGlobal");
+        BRunUtil.invoke(parallelCompileResult, "testLockWithInvokableAccessingGlobal");
     }
 
     @Test(description = "Test for parallel run when invocations have loops and global var dependencies")
     public void testParallelRunWithChainedInvocationDependencies() {
-        Object returns = BRunUtil.invoke(parallelCompileResult, "testLockWIthInvokableChainsAccessingGlobal");
+        BRunUtil.invoke(parallelCompileResult, "testLockWIthInvokableChainsAccessingGlobal");
     }
 
     @Test(description = "Test for parallel run when invocation are recursive and have global var dependencies")
     public void testParallelRunWithRecursiveInvocationDependencies() {
-        Object returns = BRunUtil.invoke(parallelCompileResult, "testLockWIthInvokableRecursiveAccessGlobal");
+        BRunUtil.invoke(parallelCompileResult, "testLockWIthInvokableRecursiveAccessGlobal");
```

Comment:
Shall we add the tests in same `bal` files to `dataProvider`s?

Reply:
Each of these tests have a description associated with them. If I move them to use a `dataProvider`, those descriptions get omitted. Hence I did not do it like that.